src/explore-education-statistics-common/src/services/util/combineMeasuresWithDuplicateLocationCodes.ts

```diff
@@ -0,0 +1,245 @@
+              location: {
+                [level]: {
+                  name: combinedLocation.label,
+                  code,
+                },
+              },
+              measures: allAvailableMeasures.reduce(
+                (acc, measure) => ({
+                  ...acc,
+                  [measure]: '',
+                }),
+                {},
+              ),
+            } as TableDataResult;
+          });
+        },
+      );
+
+      const allAvailableLocationNames = Object.keys(
+        resultsGroupedByLocationName,
+      ).sort();
+
+      // Now for each combination of Time Period and Filters, produce a single combined row of data that merges
+      // the duplicate rows from each Location into one, using a strategy to merge a set of data for each measurement
+      // into a single value.
+      const deduplicatedResults = timePeriodFilterCombinations.flatMap(
+        combination => {
+          // For each measure, collect the value for that measure from each Location's results for this Time Period and
+          // Filter combination, or undefined if a Location does not have a value that matches this criteria.
+          const measureValuesForEachLocation = Object.keys(
+            combination.measures,
+          ).reduce((acc, measure) => {
+            const measureValues = allAvailableLocationNames.map(
+              locationName => {
+                const resultForLocationName = resultsGroupedByLocationName[
+                  locationName
+                ].find(
+                  result =>
+                    result.timePeriod === combination.timePeriod &&
+                    isEqual(combination.filters, result.filters),
+                );
+                return resultForLocationName?.measures[measure];
+              },
+            );
+            return {
+              ...acc,
+              [measure]: measureValues,
+            };
+          }, {});
+
+          // Now for each measure, combine the values gathered from all of the Locations into a single value per measure.
+          const mergedMeasurements = mapValues(
+            measureValuesForEachLocation,
+            measurementsMergeStrategy,
+          );
+
+          // Return the Time Period / Filter combination, now with the merged measurement values from all of the
+          // Locations
+          return produce(combination, draft => {
+            draft.measures = mergedMeasurements;
+          });
+        },
+      );
+
+      return [...unaffectedResults, ...deduplicatedResults];
+    },
+  );
+}
+
+/**
+ * This is a strategy for merging measurement values from several duplicate Locations into a single value.
+ * This strategy produces a string with each Location's value for this measurement separated with forward slashes.
+ *
+ * ${@param measurementValues} represents the multiple values from each Location for this measurement.
+ */
+export const slashSeparatedStringMergeStrategy: MeasurementsMergeStrategy = (
```

Comment:
Is this used on the code anywhere? I can only find references to it in the tests.

Reply:
It is only used in the tests, yeah - I was originally using this strategy in the production code but it made problems for charts, that can only work off purely numeric values.  I thought it made the effects of the cell merging more obvious though in the tests for readability.  I suppose I could just move this into the test class itself if that's OK with you?

Reply:
yep, sounds good

Reply:
Moved into test class, thanks.